modules/oe_whitelabel_starter_event/oe_whitelabel_starter_event.module

```diff
@@ -186,26 +186,33 @@ function _oe_whitelabel_starter_event_preprocess_date(array &$variables): void {
     return;
   }
 
+  // Displayed date depends on the timezone.
+  $cache = CacheableMetadata::createFromRenderArray($variables);
+  $cache->addCacheContexts(['timezone']);
+  $cache->applyTo($variables);
+
   $dates = $node->get('oe_sc_event_dates')->first();
-  /** @var \Drupal\Core\Datetime\DrupalDateTime $start */
-  $start = $dates->get('start_date')->getValue();
+  $start = $dates->get('start_date')->getValue()->getTimestamp();
```

Comment:
The second way to do it may be 

```php
use Drupal\Core\Datetime\DrupalDateTime;
// ...
$start = new DrupalDateTime($dates->get('start_date')->value);
// ...
'year' => $start->format('Y'),
```

But I am not sure if it is much better

Reply:
there is also a 3rd variant with `$start->setTimeZone(timezone_open($timezone));` like done in DateTimeFormatterBase, but I don't mind using a service.